src/service/inactive-administrative-cost-service.ts

```diff
@@ -0,0 +1,320 @@
+        isNotEligible = true;
+      }
+
+      if (!isNotEligible) {
+        const response: UserToInactiveAdministrativeCostResponse = { userId: user.id };
+        eligibleUsers.push(response);
+      }
+    }
+
+    return eligibleUsers;
+  }
+
+  /**
+   * Deletes the given InactiveAdministrativeCost and creates an undo transfer
+   * @param inactiveAdministrativeCostId
+   */
+  public async deleteInactiveAdministrativeCost(inactiveAdministrativeCostId: number)
+    : Promise<InactiveAdministrativeCost | undefined> {
+    // Find base inactive administrative cost entity.
+    const inactiveAdministrativeCost = await this.manager.findOne(InactiveAdministrativeCost, { ...InactiveAdministrativeCostService.getOptions({ inactiveAdministrativeCostId }) });
+    if (!inactiveAdministrativeCost) return undefined;
+
+    // Get amount from transfer
+    const amount: DineroObjectRequest = inactiveAdministrativeCost.transfer.amountInclVat.toObject();
+
+    // We create an undo transfer that sends the money back to the person.
+    const undoTransfer: TransferRequest = {
+      amount,
+      description: 'Deletion of InactiveAdministrativeCost',
+      fromId: 0,
+      toId: inactiveAdministrativeCost.fromId,
+    };
+
+    // Save new transfer and delete the administrative cost
+    await new TransferService(this.manager).postTransfer(undoTransfer);
+    await this.manager.delete(InactiveAdministrativeCost, inactiveAdministrativeCostId);
+
+    return inactiveAdministrativeCost;
+  }
+
+  /**
+   * Creates an InactiveAdministrativeCost from an InactiveAdministrativeCostRequest
+   * @param inactiveAdministrativeCostRequest - The InactiveAdministrativeCost request to create
+   */
+  public async createInactiveAdministrativeCost(inactiveAdministrativeCostRequest: CreateInactiveAdministrativeCostRequest)
+    : Promise<InactiveAdministrativeCost> {
+    const { forId } = inactiveAdministrativeCostRequest;
+
+    // Calculate reduction amount
+    const user = await this.manager.findOne(User, { where: { id: forId } });
+    const userBalance = await new BalanceService(this.manager).getBalance(forId);
+
+    const monetaryAmount = (userBalance.amount.amount < 5) ? userBalance.amount.amount - 5 : 5;
+
+    const amount: DineroObjectRequest = {
+      amount: monetaryAmount,
+      currency: 'EUR',
+      precision: 2,
+    };
+
+    // Create transfer request and create the linked transfer
+    const transferRequest: TransferRequest = {
+      amount,
+      description: 'InactiveAdministrativeCost Transfer',
+      fromId: forId,
+      toId: 0,
+    };
+
+    const transfer = await new TransferService(this.manager).createTransfer(transferRequest);
+
+    // Create a new inactive administrative cost
+    const newInactiveAdministrativeCost: InactiveAdministrativeCost = Object.assign(new InactiveAdministrativeCost(), {
+      fromId: forId,
+      from: user,
+      amount: dinero(amount),
+      transfer: transfer,
+    });
```

Comment:
prefer `insert` or `save` as this now does not have any linting 

(I know, we do this wrong _everywhere_)

Reply:
Changed it a bit and it works but not certain what you mean.